src/components/DynamicRacingBar/DynamicRacingBar.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect, useRef } from 'react';
+import * as echarts from 'echarts';
+import * as React from 'react';
+import PlayButton from './DynamicRacingBarin/PlayButton';
+import { getGithubTheme } from '../../utils/utils';
+import { DynamicBarData } from '../../mock/DynamicRacingBar.data';
+const LIGHT_THEME = {
+  FG_COLOR: '#24292f',
+  BG_COLOR: '#ffffff',
+  PALLET: ['#5470c6', '#91cc75'],
+};
+const DARK_THEME = {
+  FG_COLOR: '#c9d1d9',
+  BG_COLOR: '#0d1118',
+  PALLET: ['#58a6ff', '#3fb950'],
+};
+const githubTheme = getGithubTheme();
+interface DynamicBarProps {
+  dataURL: string;
+  data1: [string, number][];
+  data2: [string, number][];
+}
+interface Symbol {
+  name: string;
+  avatar: string;
+}
+let resname = DynamicBarData.flags[0].name;
+let resavatar = DynamicBarData.flags[0].emoji;
+const symbols: Symbol[] = [{ name: resname, avatar: resavatar }];
+const years: string[] = [];
+function getContri(userName: string) {
+  if (!userName) {
+    return '';
+  }
+  return (
+    symbols.find(function (item) {
+      return item.name === userName;
+    }) || {}
+  ).avatar;
+}
+const DynamicRacingBar: React.FC<DynamicBarProps> = (props) => {
+  const { data1, data2 } = props;
+  const [playing, setPlaying] = useState(false);
+  const divEL = useRef(null);
+  const TH = githubTheme == 'light' ? LIGHT_THEME : DARK_THEME;
+  const data: number[] = [];
+  for (let i = 0; i < 5; ++i) {
+    data.push(Math.round(Math.random() * 200));
+  }
+  const option: echarts.EChartsOption = {
+    xAxis: {
+      max: 'dataMax',
+    },
+    yAxis: {
+      type: 'category',
+      data: ['A', 'B', 'C', 'D', 'E'],
+      inverse: true,
+      animationDuration: 300,
+      animationDurationUpdate: 300,
+    },
+    series: [
+      {
+        realtimeSort: true,
+        name: 'X',
+        type: 'bar',
+        data: data,
+        label: {
+          show: true,
+          position: 'right',
+          valueAnimation: true,
+        },
+      },
+    ],
+    legend: {
+      show: true,
+    },
+    animationDuration: 20000,
+    animationDurationUpdate: 20000,
```

Comment:
`animationDuration` here only affects how long it take to finish a cycle of animation from an option to another.

<img width="820" alt="image" src="https://user-images.githubusercontent.com/32434520/191165021-54c56059-3787-4133-9a50-f82dc736a583.png">

Currently the bars are "moving", but moving to values that were decided at first, i.e. after `setOption`, echarts knows that C will achieve 190, but it takes 20s animations to draw the bar from 0 to 190.